src/qc_ncsa.aplf

```diff
@@ -152,26 +152,52 @@
         H ← (÷2*0.5)×(2 2 ⍴ 1 1 1 ¯1)
 
         ⍝ P (phase shift)
-        P ← {2 2 ⍴ ((1 0 0),(*(0J1×⍵)))}
+        P ← {
+            ph ← *0J1×⍵
+            2 2 ⍴ ((1 0 0),ph)
+        }
 
         ⍝ T (4-th root of Z)
-        T ← 2 2 ⍴ ((1 0 0),(*(0J1×(0.25×○1))))
+        T ← 2 2 ⍴ ((1 0 0),*0J1×0.25×○1)
 
         ⍝ Rx (rotation in x)
-        Rx ← {2 2 ⍴ ((2○⍵÷2),(0J¯1×(1○⍵÷2)),(0J¯1×(1○⍵÷2)),(2○⍵÷2))}
+        Rx ← {
+            a←2○⍵÷2
+            b←0J¯1×1○⍵÷2
+            2 2 ⍴ a b b a
+        }
 
         ⍝ Ry (rotation in y)
-        Ry ← {2 2 ⍴ ((2○⍵÷2),(¯1×(1○⍵÷2)),(1○⍵÷2),(2○⍵÷2))}
+        Ry ← {
+            a ← 2○⍵÷2
+            b ← 1○⍵÷2
+            bm ← ¯1×b
+            2 2 ⍴ a bm b a
+        }
```

Comment:
what are the 1s and 0s under the values here: 
![image](https://github.com/nunezco2/quAPL/assets/38033108/8da1204b-ac65-40cc-a32a-334dd21ec615)

Reply:
is it an identity matrix?

Reply:
oh, you have to pass them the angle of the rotation? and that makes the specific gate for that specific rotation?